apps/token-manager/app/src/App.js

```diff
@@ -86,100 +98,151 @@ class App extends React.PureComponent {
+            <Button
+              mode="strong"
+              onClick={this.handleLaunchAssignTokensNoHolder}
+              css={`
+                ${layoutName === 'small'
+                  ? `
+                    width: ${5 * GU}px;
+                    height: ${5 * GU}px;
+                    min-width: 0;
+                    padding: 0;
+                  `
+                  : ''}
+              `}
             >
-              {appStateReady && (
-                <AssignVotePanelContent
-                  opened={sidepanelOpened}
-                  tokenDecimals={numData.tokenDecimals}
-                  tokenDecimalsBase={tokenDecimalsBase}
-                  onUpdateTokens={this.handleUpdateTokens}
-                  getHolderBalance={this.getHolderBalance}
-                  maxAccountTokens={maxAccountTokens}
-                  {...assignTokensConfig}
-                />
-              )}
-            </SidePanel>
-          </IdentityProvider>
-        </div>
-      </Main>
+              {layoutName === 'small' ? <IconPlus /> : 'Add tokens'}
+            </Button>
+          }
+        />
+
+        {appStateReady && holders.length > 0 ? (
+          <Holders
+            holders={holders}
+            groupMode={groupMode}
+            maxAccountTokens={maxAccountTokens}
+            tokenAddress={tokenAddress}
+            tokenDecimalsBase={tokenDecimalsBase}
+            tokenName={tokenName}
+            tokenSupply={tokenSupply}
+            tokenSymbol={tokenSymbol}
+            tokenTransfersEnabled={tokenTransfersEnabled}
+            userAccount={connectedAccount}
+            onAssignTokens={this.handleLaunchAssignTokens}
+            onRemoveTokens={this.handleLaunchRemoveTokens}
+          />
+        ) : (
+          !isSyncing && (
+            <EmptyState onActivate={this.handleLaunchAssignTokensNoHolder} />
+          )
+        )}
+
+        <SidePanel
+          title={
+            assignTokensConfig.mode === 'assign'
+              ? 'Add tokens'
+              : 'Remove tokens'
+          }
+          opened={sidepanelOpened}
+          onClose={this.handleSidepanelClose}
+          onTransitionEnd={this.handleSidepanelTransitionEnd}
+        >
+          {appStateReady && (
+            <AssignVotePanelContent
+              opened={sidepanelOpened}
+              tokenDecimals={numData.tokenDecimals}
+              tokenDecimalsBase={tokenDecimalsBase}
+              onUpdateTokens={this.handleUpdateTokens}
+              getHolderBalance={this.getHolderBalance}
+              maxAccountTokens={maxAccountTokens}
+              {...assignTokensConfig}
+            />
+          )}
+        </SidePanel>
+      </IdentityProvider>
     )
   }
 }
 
 export default () => {
   const { api, appState, connectedAccount, requestMenu } = useAragonApi()
+  const theme = useTheme()
+  const themeMode = useThemeMode()
+  const { layoutName } = useLayout()
+
+  useEffect(() => {
+    const fn = ({ data }) => {
```

Comment:
Super nitpick here: `useCallback` on the handler function

Reply:
Same than above, I’ll remove this part.